### PR TITLE
feat: zendesk chat stream stability

### PR DESCRIPTION
**Pull request by bsadecki-magi**
# Zendesk Chat Stability Enhancements

This PR implements several improvements to enhance the stability and reliability of our pub/sub and stream functionality, particularly focusing on the Zendesk integration.

### Webhook Handling
- Added webhook ID support in message payload for better traceability
- Implemented webhook configuration error handling
- Added the foundations for webhook secret validation for improved security

### Connection Management
- Implemented keep-alive interval to maintain stable connections
- Added reconnection logic with configurable interval
- Enhanced connection status handling for more reliable communication

### Type Safety
- Added new type definitions:
  - `HandoffChatMessage` for consistent message handling
  - `ZendeskMessagePayload` for type-safe webhook payloads

### Dependencies
- Added `@vercel/functions` package for improved serverless function stability
- Updated `pnpm-lock.yaml` to include new dependencies

### Configuration
- Added webhookId and webhookSecret to settings file
- Improved error handling for missing webhook configurations

## Testing
- Tested reconnection scenarios
- Validated message type safety
- Confirmed keep-alive functionality

### CODE REVIEW

File: settings.d.ts [hunk 1]
```diff
@@ -18,6 +18,8 @@ declare global {
     apiKey: string;
     apiSecret: string;
     appId: string;
+    webhookId: string;
+    webhookSecret: string;
```
Comment (jcruz2us) [inline settings.d.ts:22]: can we make these optional? Front will also rely on webhooks to receive agent messages but won't require these

Comment (bsadecki-magi) [reply settings.d.ts]: Fixed! Good callout.

File: lib/useHandoff.ts [hunk 2]
```diff
@@ -216,20 +259,31 @@ export function useHandoff({ messages, signedUserData }: HandoffProps) {
...
       }
     } catch (error) {
       if (error instanceof Error && error.name === "AbortError") {
         console.log("Fetch aborted");
       } else {
         console.error("Error streaming response:", error);
+        void handleEndHandoff();
+      }
+    } finally {
+      setIsConnected(false);
```
Comment (jcruz2us) [inline lib/useHandoff.ts:273]: shouldn't this only be set to false if there is a disconnect/in the catch block?

File: settings.d.ts [hunk 3]
```diff
@@ -13,12 +13,22 @@ declare global {
...
     appId: string;
-  };
+  } & (
+    | {
+        type: "zendesk";
+        webhookId: string;
+        webhookSecret: string;
+      }
+    | {
+        type: "salesforce" | "front";
```
Comment (jcruz2us) [inline settings.d.ts:27]: nice!

**Review by jcruz2us — APPROVED** [review]
Looks good. Left 1 additional comment but not a blocker

File: lib/useHandoff.ts (hunk 2 above)
Comment (bsadecki-magi) [reply lib/useHandoff.ts:273]: In both cases, request has finished with 200 and when request errors out, we'll need to set connected to false.

**Pull request merged**